app/src/userpages/components/StreamPage/Show/HistoryView/index.jsx

```diff
@@ -2,99 +2,258 @@
 }
 
-type Props = StateProps & DispatchProps
+type Props = OwnProps & StateProps & DispatchProps
 
 type State = {
-    range?: {
+    range: ?{
         beginDate: string,
         endDate: string,
     },
-    deleteDate?: Date,
+    deleteDate: ?Date,
+    isModalOpen: boolean,
+    csvFile: ?File,
+    confirmError: ?string,
 }
 
+const DropTarget = ({ mouseOver }: { mouseOver: boolean }) => (
+    <div className={styles.dropTarget}>
+        <div className={styles.dropTargetContent}>
+            <SvgIcon
+                name="csvUpload"
+                className={cx(styles.icon, {
+                    [styles.hover]: mouseOver,
+                })}
+            />
+            <Translate value="userpages.streams.edit.history.uploadCsv" />
+        </div>
+    </div>
+)
+
 class HistoryView extends Component<Props, State> {
-    state = {}
+    state = {
+        range: undefined,
+        deleteDate: undefined,
+        isModalOpen: false,
+        csvFile: undefined,
+        confirmError: undefined,
+    }
 
-    componentDidMount() {
-        this.load(this.props)
+    componentDidUpdate(prevProps) {
+        const { streamId } = this.props
+        const { streamId: prevStreamId } = prevProps
+
+        // Load data if stream changes
+        if (streamId && streamId !== prevStreamId) {
+            this.loadData()
+        }
     }
 
-    load({ stream }) {
-        if (stream && stream.id) {
-            getRange(stream.id).then((range) => {
+    loadData() {
+        const { getRange, streamId } = this.props
+
+        if (streamId) {
+            getRange(streamId).then((range) => {
                 this.setState({
                     range,
                 })
             }, console.error)
         }
     }
 
-    onDeleteDateChanged = (event) => {
+    onDeleteDateChanged = (date) => {
         this.setState({
-            deleteDate: event.target.value,
+            deleteDate: date,
         })
     }
 
-    deleteDataUpTo = (streamId: StreamId, date?: Date) => {
+    deleteDataUpTo = (streamId: StreamId, date: ?Date) => {
         if (date) {
             this.props.deleteDataUpTo(streamId, date)
+                .then(() => {
+                    this.loadData()
+                })
+        }
+    }
+
+    onDropAccepted = (files: Array<File>) => {
+        if (files.length <= 0) {
+            return
+        }
+
+        this.setState({
+            csvFile: files[0],
+        }, this.uploadCsv)
+    }
+
+    uploadCsv = () => {
+        const { streamId, uploadCsvFile } = this.props
+        const { csvFile } = this.state
+
+        if (streamId && csvFile) {
+            uploadCsvFile(streamId, csvFile)
+                .then(() => {
+                    // Old API would accept the file as it is but now we
+                    // have to confirm the fields
+                    this.openConfigurationModal()
+                })
+                .catch(() => {
+                    this.openConfigurationModal()
```

Comment:
This catches the error thrown in line 410 of `app/src/userpages/modules/userPageStreams/actions.js`:

```
if (response.data.schema.timestampColumnIndex == null) {
    dispatch(uploadCsvFileUnknownSchema(id, response.data.fileId, response.data.schema))
    throw new Error('Could not parse timestamp column!')
}
```
Is that desired behaviour? Maybe we should add a specific error type and catch it here and throw anything beside that?

Reply:
Yeah, It's desired behaviour but I agree that specific error type should be added 👍